asteroid/separate.py

```diff
@@ -120,50 +121,84 @@ def file_separate(
-
-            wav = resample(wav[:, 0], orig_sr=fs, target_sr=model.sample_rate)[:, None]
-        else:
-            raise RuntimeError(
-                f"Received a signal with a sampling rate of {fs}Hz for a model "
-                f"of {model.sample_rate}Hz. You can pass `resample=True` to resample automatically."
-            )
+    if resample:
+        wav = _resample(wav[:, 0], orig_sr=fs, target_sr=int(model.sample_rate))[:, None]
+    elif fs != model.sample_rate:
+        raise RuntimeError(
+            f"Received a signal with a sampling rate of {fs}Hz for a model "
+            f"of {model.sample_rate}Hz. You can pass `resample=True` to resample automatically."
+        )
     # Pass wav as [batch, n_chan, time]; here: [1, 1, time]
     wav = wav[:, 0][None, None]
-    (to_save,) = numpy_separate(model, wav, **kwargs)
+    (est_srcs,) = numpy_separate(model, wav, **kwargs)
+    # Resample to original sr
+    est_srcs = [
+        _resample(est_src, orig_sr=int(model.sample_rate), target_sr=fs) for est_src in est_srcs
+    ]
 
     # Save wav files to filename_est1.wav etc...
-    for src_idx, est_src in enumerate(to_save):
-        base = ".".join(filename.split(".")[:-1])
-        save_name = base + "_est{}.".format(src_idx + 1) + filename.split(".")[-1]
-        if output_dir is not None:
-            save_name = os.path.join(output_dir, save_name.split("/")[-1])
-        if os.path.isfile(save_name) and not force_overwrite:
-            warnings.warn(
-                f"File {save_name} already exists, pass `force_overwrite=True` to overwrite it",
-                UserWarning,
-            )
-            return
-        if fs != model.sample_rate:
-            from librosa import resample
-
-            est_src = resample(est_src, orig_sr=model.sample_rate, target_sr=fs)
-        sf.write(save_name, est_src, fs)
+    for src_idx, est_src in enumerate(est_srcs, 1):
+        sf.write(save_name_template.format(src_idx), est_src, fs)
+
+
+def _resample(wav, orig_sr, target_sr, _resamplers={}):
+    from julius import ResampleFrac
+
+    if orig_sr == target_sr:
+        return wav
+
+    # Cache ResampleFrac instance to speed up resampling if we're repeatedly
+    # resampling between the same two sample rates.
+    try:
+        resampler = _resamplers[(orig_sr, target_sr)]
+    except KeyError:
+        resampler = _resamplers[(orig_sr, target_sr)] = ResampleFrac(orig_sr, target_sr)
+
+    return resampler(torch.from_numpy(wav)).numpy()
+
+
+def _load_audio(filename):
+    try:
+        import librosa
+    except ModuleNotFoundError:
+        librosa = None
+
+    if librosa:
+        wav, sr = librosa.load(filename, dtype="float32", sr=None, duration=10)
+        # Always return wav of shape [time, n_chan]
+        if wav.ndim == 1:
+            return wav[:, None], sr
+        else:
+            return wav.T, sr
+    else:
+        return sf.read(filename, dtype="float32", always_2d=True)
```

Comment:
What is the rationale behind this? 

The only I see, supporting all the audio formats supported by librosa. 

So instead, let's try to load with soundfile directly, if it doesn't work, try to import librosa, if it doesn't work raise an error. 
Librosa has a large import time, if we can avoid it by using soundfile by default, that would be better IMO.

Reply:
Agreed, will change order of things!